tools/jvm/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSClosureOptimizer.scala

```diff
@@ -177,7 +186,10 @@ object ScalaJSClosureOptimizer {
       /** Base path to relativize paths in the source map */
       relativizeSourceMapBase: Option[URI] = None,
       /** Elements we won't warn even if they don't exist */
-      noWarnMissing: Seq[ScalaJSOptimizer.NoWarnMissing] = Nil
+      noWarnMissing: Seq[ScalaJSOptimizer.NoWarnMissing] = Nil,
+      /** Custom js code that wraps the output */
+      customOutputWrapper: (String, String) = ("", "")
+
   ) extends OptimizerConfig with ScalaJSOptimizer.OptimizerConfig
```

Comment:
That will cause binary compatibility issues. And since it's in a surface API of the tools, and we haven't broken that part of the API since 0.6.0, we should try very hard not to break binary compat.

It was of course a mistake to make `Config` a case class, but well ... We'll have to fix that in 1.0.0. In the meantime, I suggest moving the a scheme similar to what we do for [`Semantics`](https://github.com/scala-js/scala-js/blob/master/tools/shared/src/main/scala/org/scalajs/core/tools/sem/Semantics.scala), i.e.:

* `final class` with private constructor (without default values)
* add the old constructor as a deprecated `def this()`, with all its default values
* same for a deprecated `apply` method in the companion
* a non-deprecated `apply` method in the companion with only the required parameters, i.e., `output`, which calls the private constructor and explicitly sets every optional parameter to its default value.
* explicitly written `equals`, `hashCode`
* `withXYZ` methods to incrementally set all the parameters

The same applies to the other Config object in `ScalaJSOptimizer`.

Locally, you can test binary compatibility with

    > tools/mimaReportBinaryIssues